### PR TITLE
Feature: react-hook-form

### PR DESCRIPTION
**Pull request by TOPOFGR**
Details:

- Now all forms are validated by the `react-hook-form` library

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,15 +3,18 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@hookform/resolvers": "^0.1.0",
     "bulma": "^0.9.0",
     "classnames": "^2.2.6",
     "date-fns": "^2.15.0",
     "firebase": "^7.18",
+    "mutationobserver-shim": "^0.3.7",
```
Comment (MateoKruk) [inline package.json:11]: Why we use this package?

File: src/components/ErrorMessage/index.jsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { useFormatMessage } from 'hooks';
+
+const ErrorMessage = (text = '') => {
```
Comment (MateoKruk) [inline src/components/ErrorMessage/index.jsx:4]: No proptypes?

File: src/components/UserForm/index.jsx [hunk 3]
```diff
@@ -21,49 +26,41 @@ const UserForm = ({ isEditing, isProfile, user, setUser, action }) => {
 
   const dispatch = useDispatch();
 
+  const schema = yup.object().shape({
```
Comment (MateoKruk) [inline src/components/UserForm/index.jsx:29]: This schema should be outside the component.

File: src/components/UserForm/index.jsx [hunk 4]
```diff
@@ -3,15 +3,20 @@ import React, { useEffect } from 'react';
...
-import { validateEmail } from 'utils';
-import './UserForm.scss';
+import { useFormatDate, useFormatMessage } from 'hooks';
 import DatePicker from '../DatePicker';
+import errorMessage from '../ErrorMessage';
+
+import './UserForm.scss';
 
-const UserForm = ({ isEditing, isProfile, user, setUser, action }) => {
+const UserForm = ({ isEditing, isProfile, user, action }) => {
```
Comment (MateoKruk) [inline src/components/UserForm/index.jsx:19]: I'm not a big fan of `UserForm` having its own state and dispatching action. This component should be only a representational component and the state should be managed by other components like `User`.

File: src/pages/Login/index.jsx [hunk 5]
```diff
@@ -30,12 +35,15 @@ const Login = () => {
 
   const dispatch = useDispatch();
 
-  const [authData, setAuthData] = useState({
-    email: '',
-    password: '',
+  const schema = yup.object().shape({
```
Comment (MateoKruk) [inline src/pages/Login/index.jsx:38]: This schema should be outside the component.

File: src/pages/Login/index.jsx [hunk 6]
```diff
@@ -30,12 +35,15 @@ const Login = () => {
...
+    email: yup.string().email().required(),
+    password: yup.string().min(6).required(),
   });
 
-  const onChangeHandler = useChangeHandler(setAuthData);
```
Comment (MateoKruk) [inline src/pages/Login/index.jsx:38]: As I said before, we should manage our state between components like this one.

File: src/pages/Profile/ChangePassword/index.jsx [hunk 7]
```diff
@@ -1,88 +1,63 @@
...
     }),
     shallowEqual
   );
 
   const dispatch = useDispatch();
 
-  useEffect(() => {
-    return () => dispatch(authCleanUp());
-  }, [dispatch]);
+  const schema = yup.object().shape({
```
Comment (MateoKruk) [inline src/pages/Profile/ChangePassword/index.jsx:23]: This schema should be outside the component.

File: src/pages/ResetPassword/index.jsx [hunk 8]
```diff
@@ -1,28 +1,38 @@
...
-      isAuth: !!state.auth.userData.userId
+      isAuth: !!state.auth.userData.userId,
     }),
     shallowEqual
   );
 
   const dispatch = useDispatch();
 
-  const [resetPasswordData, setResetPasswordData] = useState('');
+  const schema = yup.object().shape({
```
Comment (MateoKruk) [inline src/pages/ResetPassword/index.jsx:29]: The same for this schema.

File: package.json (hunk 1 above)
Comment (TOPOFGR) [reply package.json:11]: There was an issue when testing and the react-hook-form recommend installing it, you can check it [here](https://react-hook-form.com/faqs#TestingfailedduetoMutationObserver).

**Review by MateoKruk — CHANGES_REQUESTED** [review]
**Change password form**:
- When submitting the form with the current password empty I don't get a message saying that field is required.
- While typing in the current password the field turns red and no message is shown.
- Once we successfully changed our password the fields aren't cleared automatically.


Sep-25-2020 14-44-58

**Users table**:
- When I use the search box I get an error.

Screen Shot 2020-09-25 at 14 48 54

**Login form**:
- The password input doesn't show a required message.

Screen Shot 2020-09-25 at 14 50 26

**Set password form**:
- When we enter with the email link the email password is green all the time for some reason.
- When we start typing the password the UI changes for some reason.
- When the user enters with an email link I think we should't show the option to continue with social media.

Sep-25-2020 14-54-59

File: src/pages/Login/index.jsx [hunk 9]
```diff
@@ -148,24 +143,30 @@ const Login = () => {
...
-                      {inputs.password.message && (
-                        <p
-                          className={classNames(
-                            'help',
-                            inputs.password.modifier
-                          )}
-                        >
-                          {inputs.password.message}
-                        </p>
+                      {isEmailLink ? (
```
Comment (tpiaggio) [inline src/pages/Login/index.jsx:158]: This block of code could be polished so we avoid nested ternary

File: src/pages/Profile/ChangePassword/index.jsx [hunk 10]
```diff
@@ -1,88 +1,80 @@
...
+  const currentPassword = watch('current');
+  const confirmationPassword = watch('confirmation');
 
-  const setInputs = (key, value) => {
-    inputs = { ...inputs, [`${key}`]: value };
-  };
+  const isNewPasswordSecure = newPassword && newPassword.length >= 6;
 
-  const isNewPasswordSecure = passwords.new && passwords.new.length >= 6;
+  const isCurrentPasswordInvalid =
```
Comment (tpiaggio) [inline src/pages/Profile/ChangePassword/index.jsx:61]: Can we do realtime validations with yup / react-hook-form and not wait for a submit in order to validate? if we can, we should avoid using this manual validations

File: src/pages/Users/index.jsx [hunk 11]
```diff
@@ -160,6 +160,7 @@ const Users = () => {
         delete clonedElem.id;
         delete clonedElem.isAdmin;
         delete clonedElem.logoUrl;
+        delete clonedElem.teams;
```
Comment (tpiaggio) [inline src/pages/Users/index.jsx:163]: Let's remove this

**Review by MateoKruk — APPROVED** [review]

**Pull request merged**